src/ui/map.js

```diff
@@ -436,6 +436,13 @@ class Map extends Camera {
         if (typeof window !== 'undefined') {
             window.addEventListener('online', this._onWindowOnline, false);
             window.addEventListener('resize', this._onWindowResize, false);
+            if (typeof window.screen.orientation === 'object') {
+                // ensures browser supports latest ScreenOrientation API
+                window.screen.orientation.addEventListener('change', this._onWindowResize, false);
```

Comment:
wherever an event listener is added on map create, it needs to also be removed on map remove, eg https://github.com/mapbox/mapbox-gl-js/blob/42ccea03b45d147628f261bca62c27fdace5eaa1/src/ui/map.js#L2566-L2569